### PR TITLE
WIP Fix for #4455

### PR DESCRIPTION
**Pull request by galv**
# What does this PR do ?

Proposed fix for #4455

**Collection**: asr, common

# Changelog 
- [Plumb is_tar_manifest throughout dataset code.](https://github.com/NVIDIA/NeMo/commit/b1bc3ab190cbb1e24898cd2af6ff84e2425f69d7)
- [Fix for KeyError.](https://github.com/NVIDIA/NeMo/commit/653d4d755d0132d640305017336ddbf612705aeb)
# Usage

This allow for running ASR training on tar files with keys containing "/".

is_tar_manifest needs to be plumbed through all possible parse_func's in usage today.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [X] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to #4455

### CODE REVIEW

File: nemo/collections/asr/data/audio_to_text.py [hunk 1]
```diff
@@ -696,7 +700,8 @@ def __next__(self):
                     self.current_bytes, self.current_fn = next(self.iterator)
                     self.offset_id = 0
                 else:
-                    offset_list = self.collection.mapping[self.current_fn]
+                    file_id, _ = os.path.splitext(os.path.basename(audio_filename))
```
Comment (titu1994) [inline nemo/collections/asr/data/audio_to_text.py:703]: Why is this being changed ?

File: nemo/collections/common/parts/preprocessing/manifest.py [hunk 2]
```diff
@@ -96,6 +96,9 @@ def __parse_item(line: str, manifest_file: str) -> Dict[str, Any]:
     # so we will just ignore it.
     manifest_dir = Path(manifest_file).parent
     audio_file = Path(item['audio_file'])
+    if is_tar_manifest:
+        # Don't change item['audio_file'] at all
+        pass
```
Comment (titu1994) [inline nemo/collections/common/parts/preprocessing/manifest.py:101]: Below should be elif or there is no effect of this check.

**Comment by galv** [comment]
Resolved the previous messages from @titu1994 without responding since this isn't a good approach.

Force pushed a simple commit reverting #4277 for now. More details in #4455.

Note: This is my first PR. I signed the commit, not sure if there's anything else I ought to do (e.g., to restart CI).

**Comment by titu1994** [comment]
Steve added some changes to main branch, could you try rebasing this PR with main, and then try out if his fixes were sufficient ? If not we may need further changes.

**Comment by titu1994** [comment]
For DCO, ignore it for now.

**Comment by stevehuang52** [comment]
The problem was already fixed in [this PR](https://github.com/NVIDIA/NeMo/pull/4470), closing this PR

**Pull request closed without merging**